### PR TITLE
[WX-1605] Fix Codecov Report Uploads

### PR DESCRIPTION
**Pull request by THWiseman**
The [old way](https://docs.codecov.com/docs/about-the-codecov-bash-uploader) of uploading coverage repots to CodeCov has broken. Using [this github action](https://github.com/marketplace/actions/codecov) is now the preferred method. 

This branch also required [adding a secret to this repository](https://docs.codecov.com/docs/adding-the-codecov-token). I got the token from the [codecov settings page](https://app.codecov.io/gh/broadinstitute/).

### CODE REVIEW

**Review by rsaperst — APPROVED** [review]

**Review by aednichols — APPROVED** [review]
Works for me if tests pass!

File: .github/workflows/integration_tests.yml [hunk 1]
```diff
@@ -110,6 +110,13 @@ jobs:
         set -e
         echo Running test.sh
         ./src/ci/bin/test.sh
+    # This uploads the coverage report to codecov.
+    # Coverage reports are generated by SBT in src/ci/bin/test.inc.sh cromwell::private::generate_code_coverage()
+    - uses: codecov/codecov-action@v4
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
+        fail_ci_if_error: true # Tolerate missing codecov reports, since not all suites generate them.
```
Comment (aednichols) [inline .github/workflows/integration_tests.yml:118]: I think we want `false` here, the no-coverage suites are failing.

Comment (THWiseman) [reply .github/workflows/integration_tests.yml]: Trying to figure out if there's a way to fail loudly if the upload fails, but also tolerate missing code-coverage for certain suites. The `handle_no_reports_found` flag does not work!

Comment (THWiseman) [reply .github/workflows/integration_tests.yml]: Giving up on this for now. Will just leave `fail_ci_if_error` as `false`.

**Pull request merged**